pktoken/pktoken_test.go

```diff
@@ -56,3 +77,58 @@ func TestPKTokenJSON(t *testing.T) {
 
 	require.JSONEq(t, string(pktJson), string(tokenJson))
 }
+
+func generateMockPKToken(signingKey crypto.Signer, jwkKey jwk.Key) (*pktoken.PKToken, error) {
+	cic, err := clientinstance.NewClaims(jwkKey, map[string]any{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Calculate our nonce from our cic values
+	nonce, err := cic.Commitment()
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate mock id token
+	idToken, err := generateMockIDToken(nonce, "https://github.com/openpubkey", "me")
+	if err != nil {
+		return nil, err
+	}
+
+	// Sign mock id token payload with cic headers
+	cicToken, err := cic.Sign(signingKey, jwkKey.Algorithm(), idToken)
+	if err != nil {
+		return nil, err
+	}
+
+	// Combine two tokens into a PK Token
+	return pktoken.New(idToken, cicToken)
+}
+
+func generateMockIDToken(nonce, issuer, audience string) ([]byte, error) {
+	token := openid.New()
+
+	token.Set(`nonce`, nonce)
+
+	// Required token payload values for OpenID
+	token.Set(jwt.IssuerKey, issuer)
+	token.Set(jwt.AudienceKey, audience)
+	token.Set(jwt.IssuedAtKey, time.Now().Unix())
+	token.Set(jwt.ExpirationKey, time.Now().Add(24*time.Hour).Unix())
+	token.Set(jwt.SubjectKey, "1234567890")
+
+	alg := jwa.KeyAlgorithmFrom("RS256")
+	signingKey, err := util.GenKeyPair(alg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sign the token with the secret key
+	signedToken, err := jwt.Sign(token, jwt.WithKey(alg, signingKey))
+	if err != nil {
+		return nil, err
+	}
+
+	return signedToken, nil
```

Comment:
It would be nice to have some unittests that use known good ID Tokens from google. That way if our integration tests with Google start failing we have a past ID Token that we know worked and we can see what the failure is. Also really help from an understanding point of view to be able to point to a real ID Token from a particular OP.

Reply:
I would prefer to keep the provider-specific tests more with the provider and keep these tests a lot more agnostic. Essentially, I don't disagree but I don't think this is the right PR or place for that kind of test.